### PR TITLE
Peerplays transfer updates for ui support

### LINKED ISSUE #35: (S) Debug sign/transfer within ScatterBridge
Working in unit tests. Does not work in the UI. Need to find out why.

### PR DESCRIPTION
**Pull request by MuffinLightning**
Should resolve https://github.com/peerplays-network/getscatter-walletpack/issues/35

### CODE REVIEW

File: packages/peerplays/lib/peerplays.js [hunk 1]
```diff
@@ -306,19 +306,23 @@ export default class PPY extends Plugin {
     // Get the transaction
     let transferTransaction = await _PPY.getTransferTransaction(from, to, amount, memo, '1.3.0');
 
+    let privateActiveKey = await KeyPairService.publicToPrivate(account.publicKey);
```
Comment (mseaward) [inline packages/peerplays/lib/peerplays.js:309]: you're taking the account public key and calling it private key?

File: packages/core/lib/services/blockchain/TransferService.js [hunk 2]
```diff
@@ -55,7 +60,8 @@ export default class TransferService {
 		    case Blockchains.EOSIO: return transfer.transaction_id;
 		    case Blockchains.TRX: return transfer.txID;
 		    case Blockchains.ETH: return transfer.transactionHash;
-		    case Blockchains.BTC: return transfer.txid;
+            case Blockchains.BTC: return transfer.txid;
+            case Blockchains.PPY: return transfer;
```
Comment (mseaward) [inline packages/core/lib/services/blockchain/TransferService.js:64]: what exactly is returned here?

**Review by mseaward — CHANGES_REQUESTED** [review]

File: packages/core/lib/services/blockchain/TransferService.js (hunk 2 above)
Comment (MuffinLightning) [reply packages/core/lib/services/blockchain/TransferService.js:64]: `baseTransfer` in this case calls `PluginRepository.plugin` (accessing the respective blockchain plugin, in our case peerplays), and calls transfer on it. Most of these other chains have their transfer() returning an object which is then parsed. In our case, we're returning the hash directly (as mentioned in our convo), so I can just return the value of `transfer`.

File: packages/peerplays/lib/peerplays.js (hunk 1 above)
Comment (MuffinLightning) [reply packages/peerplays/lib/peerplays.js:309]: I'm taking the public key and retrieving the associated key within the keypair needed to sign the transaction.

**Review by mseaward — APPROVED** [review]

**Pull request merged**